notebooks/ch16/classification_precision_recall.md

```diff
@@ -0,0 +1,326 @@
+    <tr>
+      <th>...</th>
+      <td>...</td>
+      <td>...</td>
+    </tr>
+    <tr>
+      <th>997</th>
+      <td>&lt;html&gt;\n &lt;head&gt;\n     &lt;meta http-equiv="Conten...</td>
+      <td>1</td>
+    </tr>
+    <tr>
+      <th>998</th>
+      <td>&lt;html&gt;\n &lt;head&gt;\n &lt;/head&gt;\n &lt;body&gt;\n \n &lt;cente...</td>
+      <td>1</td>
+    </tr>
+    <tr>
+      <th>999</th>
+      <td>\n &lt;html&gt;\n \n &lt;head&gt;\n &lt;meta http-equiv=3D"Co...</td>
+      <td>1</td>
+    </tr>
+  </tbody>
+</table>
+<p>1000 rows × 2 columns</p>
+</div>
+
+
+
+Each row contains the body of an email and a `spam` column, which is `0` if the email is ham or `1` if it is spam.
+
+Suppose we have a list of words that we believe are common in spam emails: "please", "click", "money", and "business". We use it to build a feature matrix where the $i$'th column of the matrix contains a 1 if the $i$'th word in our words list is present in the email and 0 otherwise. For example, using our four chosen words and the email body "please respond by tomorrow", the corresponding row of the feature matrix would be $[1, 0, 0, 0]$.
+
+The following code trains the classifier `words_list_model`. The setup of the $X$ feature matrix and the train-test split are omitted for brevity.
+
+
+```python
+words = ['please', 'click', 'money', 'business']
+
+# HIDDEN
+
+X = pd.DataFrame(words_in_texts(words, emails['email'].str.lower())).as_matrix()
+y = emails['spam'].as_matrix()
+
+# Train-test split
+X_train, X_test, y_train, y_test = train_test_split(
+    X, y, random_state=41
+)
+
+# SHOW
+
+#Fit the model
+spam_detector = LogisticRegression(fit_intercept=True)
+spam_detector.fit(X_train, y_train)
+
+print(f'Test set accuracy: {spam_detector.score(X_test, y_test)}')
+```
+
+    Test set accuracy: 0.944
+
+
+Using just the occurrence of four words, we have built a model that accurately classifies about 94% of the emails in the test set, which at first glance is a good result. However, if we take a closer look at our data we realize that our classifier isn't performing as well as we thought. 
+
+The accuracy we computed is the fraction of predictions that the model correctly classified. The fact that the accuracy is high implies that most of the model's classifications are true positives and true negatives, which occur when the model correctly labels the positive class (spam) and negative class (ham) respectively. On the other hand, there are two types of errors a model can make:
+
+* False positive: a ham email gets flagged as spam and filtered out of the inbox. Generally, a false positive occurs when the model assigns a data point to the positive class (spam) whereas it actually belonged in the negative class (ham).
+* False negative:  a spam email gets mislabelled as ham and ends up in the inbox. Generally, a false negative occurs when the model assigns a data point to the negative class (ham) whereas it actually belonged in the positive class (spam).
```

Comment:
(nit) Supposedly, using one "l" in "mislabeled" is the "American" way of writing it.